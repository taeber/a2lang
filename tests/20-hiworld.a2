use [
    INIT: sub @ $FB2F,  ; AKA TEXT
    HOME: sub @ $FC58,
    COUT: sub <- [ch: char @ A] @ $FDED,
]

let main = sub {
    INIT()
    HOME()

    COUT(`H)
    COUT(`i)
    COUT(`.)
}

