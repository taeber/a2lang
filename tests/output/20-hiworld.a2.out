INIT	EQU $FB2F
HOME	EQU $FC58
COUT	EQU $FDED
main	JSR INIT
	JSR HOME
* COPYBB @A #"H"
	LDA #"H"
	JSR COUT
* COPYBB @A #"i"
	LDA #"i"
	JSR COUT
* COPYBB @A #"."
	LDA #"."
	JSR COUT
main._end	RTS
